source/agora/consensus/EnrollmentManager.d

```diff
@@ -687,24 +699,26 @@ unittest
     import std.conv;
 
     scope storage = new TestUTXOSet;
-
+    UTXOSetValue[Hash] utxos;
     auto gen_key_pair = WK.Keys.Genesis;
     KeyPair key_pair = KeyPair.random();
 
     foreach (idx; 0 .. 8)
     {
         auto input = Input(hashFull(GenesisTransaction), idx.to!uint);
-
+        auto output = Output(Amount.MinFreezeAmount, key_pair.address);
         Transaction tx =
         {
             TxType.Freeze,
             [input],
-            [Output(Amount.MinFreezeAmount, key_pair.address)]
+            [output]
         };
-
-        auto signature = gen_key_pair.secret.sign(hashFull(tx)[]);
+        auto tx_hash = hashFull(tx);
+        auto signature = gen_key_pair.secret.sign(tx_hash[]);
         tx.inputs[0].signature = signature;
         storage.put(tx);
+        auto ut_hash = UTXOSetValue.getHash(tx_hash, 0);
+        utxos[ut_hash] = UTXOSetValue(1, TxType.Freeze, output);
```

Comment:
This creates an invalid state. The `UTXOSetValue.unlock_height` shouldn't be `1`.